node/src/components/consensus/era_supervisor.rs

```diff
@@ -1171,6 +1171,14 @@ async fn execute_finalized_block<REv>(
 ) where
     REv: From<StorageRequest> + From<ControlAnnouncement> + From<ContractRuntimeRequest>,
 {
+    // if the block exists in storage, it either has been executed before, or we fast synced to a
+    // higher block - skip execution
+    if effect_builder
+        .check_block_header_existence(finalized_block.height())
+        .await
+    {
+        return;
+    }
```

Comment:
Do you think it's worth adding a test for the added behavior?

Reply:
It's just a nit, but I think this function:
```
async fn execute_finalized_block<REv>(...)
```
could be renamed to
```
async fn exqueue_finalized_block_for_execution<REv>(...)
```
since the real execution happens in different place.

Reply:
Could be worth adding a test, although I'm not sure how, to be honest. Sounds like it would be fairly complex (although maybe I'm just not aware about some of the testing facilities we have?).

As for the name, I'd leave it as is. I see the enqueueing part as an implementation detail - from the consensus side, what's important is that the block will get executed. Whether it's right away, or after putting it in some kind of a queue, is irrelevant to the code, so I'd prefer not to include it in the name.